### PR TITLE
Refactor: Use Standard OpenRewrite Recipes to implement Slf4j logging automatically

### PR DESCRIPTION
**Pull request by mcook42**
## Description

Use OpenRewrite to replace direct system calls with structured logging. This will significantly improve maintainers' ability to understand and debug program execution. The only manual edits were made in the tests. All other code changes were automatically performed by their respective [OpenRewrite Recipes](https://docs.openrewrite.org/) using the [Maven plugin](https://docs.openrewrite.org/running-recipes/getting-started#step-2-add-rewrite-maven-plugin-or-rewrite-gradle-plugin-to-your-project). 

## How

1. Removed the two overloaded methods `logWithDate` and inlined their `System.out.println` calls (necessary prep for the next step)
2.  Run https://docs.openrewrite.org/recipes/java/logging/systemprinttologging to replace system calls with logger calls
3. Run https://docs.openrewrite.org/recipes/java/migrate/lombok/log/useslf4j to use `@Slf4j` annotations instead of explicit properties
4. Run https://docs.openrewrite.org/recipes/java/logging/slf4j/slf4jbestpractices to implement best practices for Slf4j
5. Tweak existing unit tests that relied on log inspection.

### CODE REVIEW

File: cv-data-tasks/src/test/java/com/trihydro/tasks/actions/ValidateTmddTest.java [hunk 1]
```diff
@@ -327,8 +324,6 @@ public void validateTmdd_run_emailSendError() throws Exception {
         uut.run();
 
         // Assert
-        verify(mockUtility, times(3)).logWithDate(logMessageCaptor.capture(), eq(ValidateTmdd.class));
-        Assertions.assertEquals("Error sending summary email:", logMessageCaptor.getAllValues().get(1));
-        Assertions.assertEquals("Failed to send error summary email:", logMessageCaptor.getAllValues().get(2));
+        // TODO: how to assert behavior without inspecting logs
```
Comment (mcook42) [inline cv-data-tasks/src/test/java/com/trihydro/tasks/actions/ValidateTmddTest.java:327]: Note: this TODO is intentional. The value of this automated refactor is to get more useful and detailed logs for debugging and system validation. The test in question is highly coupled to logging to the point that verification of behaviors is difficult without inspecting the logs directly. The value of the refactor was determined to greatly outweigh the value of this edge case test verification. Instead of removing the test entirely, I disabled it with a note for future maintainers about why it's disabled. Should we find in the future that this test case is important, the structure is there and available to whomever decides to reimplement the test.
